src/main/java/decodes/tsdb/algo/BigAdder.java

```diff
@@ -1,123 +1,154 @@
 package decodes.tsdb.algo;
 
-import java.util.Date;
-
-import ilex.var.NamedVariableList;
-import ilex.var.NamedVariable;
-import decodes.tsdb.DbAlgorithmExecutive;
 import decodes.tsdb.DbCompException;
-import decodes.tsdb.DbIoException;
-import decodes.tsdb.VarFlags;
-
-//AW:IMPORTS
-//AW:IMPORTS_END
+import ilex.var.NamedVariable;
+import org.opendcs.annotations.algorithm.Algorithm;
+import org.opendcs.annotations.algorithm.Input;
+import org.opendcs.annotations.algorithm.Output;
 
-//AW:JAVADOC
 /**
-Takes up to 20 input values labeled input1 ... input20. Multiplies
-them by coefficients supplied in properties coeff1 ... coeff20.
-Adds them together and produces a single output labeled 'output'.
-The property input_constant is added to the ouput result.
-input_constant defaults to 0 if not supplied
-Values not assigned by computation are ignored.
-All coefficients default to 1.0 if not supplied.
- 
 This algorithm is a modification of Scaler adder
 Created by M. Bogner   May 2009
  */
-//AW:JAVADOC_END
+@Algorithm(
+		description ="Takes up to 20 input values labeled input1 ... input20. Multiplies\n" +
+				"them by coefficients supplied in properties coeff1 ... coeff20.\n" +
+				"Adds them together and produces a single output labeled 'output'.\n" +
+				"The property input_constant is added to the ouput result.\n" +
+				"input_constant defaults to 0 if not supplied\n" +
+				"Values not assigned by computation are ignored.\n" +
+				"All coefficients default to 1.0 if not supplied." )
 public class BigAdder extends decodes.tsdb.algo.AW_AlgorithmBase
 {
-//AW:INPUTS
```

Comment:
Does the AlgoEdit tool support annotations now, and handle code concatenations without these comments?

Reply:
No, nor am I inclined to do so. What these do do is make it easier to deal with the algorithms in a normal IDE.

If people really push for it, we can make improvements, but for creating new algorithms, VSCode and maybe a simple extension to help is likely to be a lot better of an option.

Reply:
I guess I missed the deprecation of the algoedit tool. It's not the first time that a federal agency wanted a simplified interface for coding water resource algorithms then never used it.